resources/assets/js/annotations/components/annotationCanvas.vue

```diff
@@ -713,19 +714,25 @@ export default {
                 this.modifyInteraction.setActive(defaultMode);
             }
         },
+        canAdd(state) {
+            if (!state) {
+                this.resetInteractionMode();
+            }
+        },
         canModify(state) {
-            this.modifyInteraction.setActive(state && this.isDefaultInteractionMode);
-
+            if (!state) {
+                this.resetInteractionMode();
+            } else {
+                this.modifyInteraction.setActive(state && this.isDefaultInteractionMode);
+            }
         },
         canDelete(state) {
-            if (state) {
+            if (!state) {
+                this.resetInteractionMode();
+            } else {
                 Keyboard.on('Delete', this.deleteSelectedAnnotations, 0, this.listenerSet);
                 Keyboard.on('Backspace', this.deleteLastCreatedAnnotation, 0, this.listenerSet);
             }
-            else {
-                Keyboard.off('Delete', this.deleteSelectedAnnotations, 0, this.listenerSet);
-                Keyboard.off('Backspace', this.deleteLastCreatedAnnotation, 0, this.listenerSet);
-            }
```

Comment:
Ah, I thought you meant that resetInteractionMode() would handle this. But why not turn off the other keyboard listeners in the created methods?

Reply:
> But why not turn off the other keyboard listeners in the created methods?

Because their listening methods ignore the event in case the required ability (`can*`) is not true. At least they should :smiley: